src/editors/object.js

```diff
@@ -816,22 +819,42 @@ JSONEditor.defaults.editors.object = JSONEditor.AbstractEditor.extend({
         self.addObjectProperty(i);
         editor.setValue(value[i],initial);
       }
+
       // Otherwise, remove value unless this is the initial set or it's required
       else if(!initial && !self.isRequired(editor)) {
-        self.removeObjectProperty(i);
+        editor.setValue(editor.getEmpty(),initial);
+        //self.removeObjectProperty(i);
       }
+
       // Otherwise, set the value to the default
       else {
         editor.setValue(editor.getDefault(),initial);
       }
     });
 
+    window.console.log('value:' + JSON.stringify(value, null, 4));
+    window.console.log('schemaProperties:' + JSON.stringify(this.schema.properties, null, 4));
+
     $each(value, function(i,val) {
       if(!self.cached_editors[i]) {
         self.addObjectProperty(i);
         if(self.editors[i]) self.editors[i].setValue(val,initial);
       }
     });
+/*
+    $each(this.schema.properties, function(i,prop) {
+      if(!self.cached_editors[i]) {
+        if (value[i]){
+          self.addObjectProperty(i);
+          if(self.editors[i]) self.editors[i].setValue(value[i],initial);
+        } else {
+          window.console.log('try adding:' + i);
+        }
+      }
+    });
+*/
+
+    // something that add the missing schema properties
```

Comment:
Should we remove this comment? Is this part ready?

Reply:
Yes, if we don't want to go back to the state where it only shows up fields that are present in data, we can remove those comments